File: server/Models/Activity.java
package tailwind.mail.models;

import java.time.OffsetDateTime;
import dapper.Table;

@Table(name = "activity", schema = "mail")
public class Activity {
    private Integer id;
    private Integer contactId;
    private String key;
    private String description;
    private OffsetDateTime createdAt;

    public Activity() {
        this.key = java.util.UUID.randomUUID().toString();
        this.createdAt = OffsetDateTime.now();
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getContactId() {
        return contactId;
    }

    public void setContactId(Integer contactId) {
        this.contactId = contactId;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public OffsetDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
    }
}

File: go/models/activity.go
package models

import (
	"time"
)

// Activity represents a tracking event in the system
type Activity struct {
	ID          int       `json:"id" db:"id"`
	ContactID   int       `json:"contact_id" db:"contact_id"`
	Key         string    `json:"key" db:"key"`
	Description string    `json:"description" db:"description"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
}

// NewActivity creates a new activity record
func NewActivity(contactID int, key, description string) *Activity {
	return &Activity{
		ContactID:   contactID,
		Key:         key,
		Description: description,
		CreatedAt:   time.Now().UTC(),
	}
}
